src/cryptonote_core/service_node_list.cpp

```diff
@@ -326,6 +328,120 @@ namespace service_nodes
     return result;
   }
 
+  validate_contributor_args_result validate_contributor_args(uint8_t hf_version, contributor_args_t const &contributor_args, uint64_t const *expiration_timestamp, crypto::public_key const *service_node_key, crypto::signature const *signature)
```

Comment:
Functions should really do one logical concept, but this function feels like it does two independent things: Is this function meant to validate contributor arguments, or is it meant to check a signature?  Those feel sufficiently different to me to warrant having two functions for them.  (The nullability of the arguments, which basically enables or disables half the function, kind of says the same thing).